packages-next/keystone/src/lib/id-field.ts

```diff
@@ -0,0 +1,122 @@
+import path from 'path';
+import {
+  fieldType,
+  FieldTypeFunc,
+  IdFieldConfig,
+  legacyFilters,
+  orderDirectionEnum,
+  ScalarDBField,
+  schema,
+} from '@keystone-next/types';
+import { validate } from 'uuid';
+import { isCuid } from 'cuid';
+
+const views = path.join(
+  path.dirname(require.resolve('@keystone-next/keystone/package.json')),
+  '___internal-do-not-use-will-break-in-patch/admin-ui/id-field-view'
+);
+
+const idParsers = {
+  autoincrement(val: string | null) {
+    if (val === null) {
+      throw new Error('Only an integer can be passed to id filters');
+    }
+    const parsed = parseInt(val);
+    if (Number.isInteger(parsed)) {
+      return parsed;
+    }
+    throw new Error('Only an integer can be passed to id filters');
+  },
+  cuid(val: string | null) {
+    // isCuid is just "it's a string and it starts with c"
+    // https://github.com/ericelliott/cuid/blob/215b27bdb78d3400d4225a4eeecb3b71891a5f6f/index.js#L69-L73
+    if (typeof val === 'string' && isCuid(val)) {
+      return val;
```

Comment:
Note not lower-casing here is a difference from what we discussed. My thinking is that isCuid only accepts strings that with `c`(as in  only lower-case `c`) so treating it as case insensitive seems strange.